sources/api/pluto/src/api.rs

```diff
@@ -1,87 +1,116 @@
-use serde::{Deserialize, Serialize};
+use bottlerocket_settings_models::{AwsSettingsV1, KubernetesSettingsV1, NetworkSettingsV1};
+use serde::Deserialize;
 use snafu::{ensure, ResultExt, Snafu};
 use std::ffi::OsStr;
 use tokio::process::Command;
 
 /// The result type for the [`api`] module.
 pub(super) type Result<T> = std::result::Result<T, Error>;
 
-#[derive(Serialize, Deserialize)]
-#[serde(rename_all = "kebab-case")]
-pub(crate) struct AwsK8sInfo {
-    #[serde(skip)]
-    pub(crate) region: Option<String>,
-    #[serde(skip)]
-    pub(crate) https_proxy: Option<String>,
-    #[serde(skip)]
-    pub(crate) no_proxy: Option<String>,
-    #[serde(default, skip_serializing_if = "Option::is_none")]
-    pub(crate) cluster_name: Option<String>,
-    #[serde(default, skip_serializing_if = "Option::is_none")]
-    pub(crate) cluster_dns_ip: Option<bottlerocket_modeled_types::KubernetesClusterDnsIp>,
-    #[serde(default, skip_serializing_if = "Option::is_none")]
-    pub(crate) node_ip: Option<String>,
-    #[serde(default, skip_serializing_if = "Option::is_none")]
-    pub(crate) max_pods: Option<u32>,
-    #[serde(default, skip_serializing_if = "Option::is_none")]
-    pub(crate) provider_id: Option<String>,
-    #[serde(default, skip_serializing_if = "Option::is_none")]
-    pub(crate) hostname_override: Option<String>,
-    #[serde(default, skip_serializing_if = "Option::is_none")]
-    pub(crate) hostname_override_source:
-        Option<bottlerocket_modeled_types::KubernetesHostnameOverrideSource>,
+/// A mutable view of API settings
+///
+/// `SettingsViewDelta` keeps track of all changes in a separate structure so that only the changed
+/// set can be sent back as writes to the API server.
+///
+/// For convenience, `settings_view_get!` and `settings_view_set!` macros can be used to handle
+/// the nested optional values present in the structure succinctly.
+///
+/// `settings_view_get!` also automatically attempts to read from the settings delta before falling
+/// back to the readonly settings.
+#[derive(Debug, Clone, PartialEq)]
+pub struct SettingsViewDelta {
+    readonly: SettingsView,
+    delta: SettingsView,
 }
 
-#[derive(Serialize, Deserialize)]
-#[serde(rename_all = "kebab-case")]
-pub(crate) struct AwsInfo {
-    #[serde(default, skip_serializing_if = "Option::is_none")]
-    pub(crate) region: Option<String>,
-}
+impl SettingsViewDelta {
+    pub fn from_api_response(readonly: SettingsView) -> Self {
+        Self {
+            readonly,
+            delta: SettingsView::default(),
+        }
+    }
+
+    pub fn read(&self) -> &SettingsView {
+        &self.readonly
+    }
```

Comment:
nit: maybe this? `read` sounds like it will read all settings, and someone not using the macros might call it by mistake:
```suggestion
    pub fn initial(&self) -> &SettingsView {
        &self.initial
    }
```